src/components/radio-group/radio-group.spec.ts

```diff
@@ -194,6 +195,138 @@ describe('Radio Group Component', () => {
         });
       });
     });
+
+    describe('Form integration', () => {
+      let form: HTMLFormElement;
+      let formData: FormData;
+
+      function setFormListener() {
+        form.addEventListener('submit', (event: SubmitEvent) => {
+          event.preventDefault();
+          formData = new FormData(form);
+        });
+      }
+
+      describe('Initial checked state', () => {
+        it('initial checked state through group', async () => {
+          form = await fixture(html`
+            <form>
+              <igc-radio-group name="fruit" value="orange">
+                <igc-radio value="apple">Apple</igc-radio>
+                <igc-radio value="banana">Banana</igc-radio>
+                <igc-radio value="orange">Orange</igc-radio>
+              </igc-radio-group>
+            </form>
+          `);
+          radios = Array.from(form.querySelectorAll(IgcRadioComponent.tagName));
+          setFormListener();
+
+          expect(last(radios).checked).to.be.true;
+
+          form.requestSubmit();
+          expect(formData.get('fruit')).to.equal(last(radios).value);
+        });
+
+        it('initial checked state through radio attribute', async () => {
+          form = await fixture(html`
+            <form>
+              <igc-radio-group name="fruit">
+                <igc-radio value="apple" checked>Apple</igc-radio>
+                <igc-radio value="banana">Banana</igc-radio>
+                <igc-radio value="orange">Orange</igc-radio>
+              </igc-radio-group>
+            </form>
+          `);
+          group = form.querySelector(IgcRadioGroupComponent.tagName)!;
+          radios = Array.from(form.querySelectorAll(IgcRadioComponent.tagName));
+          setFormListener();
+
+          expect(first(radios).checked).to.be.true;
+          expect(group.value).to.equal(first(radios).value);
+
+          form.requestSubmit();
+          expect(formData.get('fruit')).to.equal(first(radios).value);
+        });
+
+        it('initial multiple checked state through radio attribute', async () => {
+          form = await fixture(html`
+            <form>
+              <igc-radio-group name="fruit">
+                <igc-radio value="apple" checked>Apple</igc-radio>
+                <igc-radio value="banana" checked>Banana</igc-radio>
+                <igc-radio value="orange" checked>Orange</igc-radio>
+              </igc-radio-group>
+            </form>
+          `);
+          group = form.querySelector(IgcRadioGroupComponent.tagName)!;
+          radios = Array.from(form.querySelectorAll(IgcRadioComponent.tagName));
+          setFormListener();
+
+          // The first checked member of the group takes over
+          expect(first(radios).checked).to.be.true;
```

Comment:
Don't we usually do the last? As in the last command to request selection overrides the previous states